data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java

```diff
@@ -122,9 +120,11 @@ private Future<Void> reconcileEgress(final DataPlaneContract.Contract contract)
       final var newEgress = entry.getKey();
       final var newResource = entry.getValue();
 
-      final var oldResource = this.cachedResources.get(newResource.getUid());
+      final var cachedEgress = this.cachedEgresses.get(uid);
```

Comment:
since we cache - do we expect some expiration?

Reply:
No, this expires whenever an object is deleted or updated, why do we need expiration?